public/content/en/vibed/deploy-on-heroku.md

```diff
@@ -0,0 +1,188 @@
+# Deploy on Heroku
+
+### Pre-requirements
+
+- Heroku [account]((https://signup.heroku.com/login))
+- [Git](https://git-scm.com/) installed
+- Your application should compile without error using dmd (`dub build --build=release`). 
+
+### 1 : Setup the app
+
+The first thing to do before deploying is binding the app's port to heroku's one. 
+Heroku sets the `PORT` variable that you are supposed to bind, and listens on port 80 by default.
+
+```d
+shared static this() {
+  // ...
+  auto settings = new HTTPServerSettings;
+  // Provide a default port in case of the $PORT variable isn't set.  
+  settings.port = environment.get("PORT", "8080").to!ushort;
+  listenHTTP(settings, router);
+}
+```
+
+Also create a `Procfile`, which is a text file in the root directory of the application, which explicitly declare what command 
+should be executed to start the app.
+
+The Procfile in the example app looks like this:
+
+```
+web: ./hello-world
+```
+
+### 2 : Prepare the app 
+
+Before going further login to the heroku cli.
+
+To login use the [heroku toolbelt](https://toolbelt.heroku.com/standalone).
+
+This provides access to the Heroku Command Line Interface (CLI), which can be used for managing and scaling your applications and add-ons.
+
+After installing the toolbet just run the following
+
+```
+$ heroku login
+```
+
+### 3 : Create the app 
+
+To do so, go on the [heroku dashboard](https://dashboard.heroku.com) and create a new app. 
+After doing this memorize the name of the created app, it will be useful later. 
+
+or use the command line like this 
+
+```
+$ heroku create
+Creating app... done, ⬢ rocky-hamlet-67506
+https://rocky-hamlet-67506.herokuapp.com/ | https://git.heroku.com/rocky-hamlet-67506.git
+```
+
+The app's name here is *rocky-hamlet-67506*. 
+
+### Deploy using git 
+
+To deploy the app directly use `git` commands. A separate git remote endpoint should be added to which new releases can be pushed.
+
+As shown in the previous section, our name's app is rocky-hamlet-67506. But change it to yours. 
+
+```
+$ heroku git:remote -a rocky-hamlet-67506
+```
+
+Notice the remote endpoint is added to the git config
+
+```
+$ git remote -v
+heroku	https://git.heroku.com/rocky-hamlet-67506.git (fetch)
+heroku	https://git.heroku.com/rocky-hamlet-67506.git (push)
+```
+
+### Adding the buildpack
+
+Buildpacks are responsible for transforming deployed code into a slug, 
+which can then be executed on a dyno. Buildpacks are composed of a set of scripts, 
+and depending on the programming language, the scripts will retrieve dependencies, 
+output generated assets or compiled code, and more.
```

Comment:
I guess just compile source code is necessary, the rest is actually not accurate